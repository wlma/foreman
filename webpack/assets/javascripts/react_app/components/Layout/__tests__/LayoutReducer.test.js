import {
  LAYOUT_INITIALIZE,
  LAYOUT_SHOW_LOADING,
  LAYOUT_HIDE_LOADING,
  LAYOUT_COLLAPSE,
  LAYOUT_EXPAND,
} from '../LayoutConstants';

import reducer from '../LayoutReducer';

import { testReducerSnapshotWithFixtures } from '../../../common/testHelpers';

const fixtures = {
  'should return the initial state': {},
  'should handle LAYOUT_INITIALIZE': {
    action: {
      type: LAYOUT_INITIALIZE,
      payload: {
        items: 'some-items',
        isCollapsed: true,
        organization: 'some organization',
        location: 'some location',
      },
    },
  },
  'should handle LAYOUT_SHOW_LOADING': {
    action: {
      type: LAYOUT_SHOW_LOADING,
    },
  },
  'should handle LAYOUT_HIDE_LOADING': {
    action: {
      type: LAYOUT_HIDE_LOADING,
    },
  },
  'should handle LAYOUT_COLLAPSE': {
    action: {
      type: LAYOUT_COLLAPSE,
    },
  },
  'should handle LAYOUT_EXPAND': {
    action: {
      type: LAYOUT_EXPAND,
    },
  },
};

describe('Layout reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
