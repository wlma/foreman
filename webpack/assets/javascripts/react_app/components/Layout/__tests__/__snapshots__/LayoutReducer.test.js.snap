// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout reducer should handle LAYOUT_COLLAPSE 1`] = `
Object {
  "isCollapsed": true,
  "isLoading": false,
  "items": Array [],
}
`;

exports[`Layout reducer should handle LAYOUT_EXPAND 1`] = `
Object {
  "isCollapsed": false,
  "isLoading": false,
  "items": Array [],
}
`;

exports[`Layout reducer should handle LAYOUT_HIDE_LOADING 1`] = `
Object {
  "isCollapsed": false,
  "isLoading": false,
  "items": Array [],
}
`;

exports[`Layout reducer should handle LAYOUT_INITIALIZE 1`] = `
Object {
  "currentLocation": "some location",
  "currentOrganization": "some organization",
  "isCollapsed": true,
  "isLoading": false,
  "items": "some-items",
}
`;

exports[`Layout reducer should handle LAYOUT_SHOW_LOADING 1`] = `
Object {
  "isCollapsed": false,
  "isLoading": true,
  "items": Array [],
}
`;

exports[`Layout reducer should return the initial state 1`] = `
Object {
  "isCollapsed": false,
  "isLoading": false,
  "items": Array [],
}
`;
